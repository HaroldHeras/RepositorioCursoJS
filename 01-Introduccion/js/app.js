
//Pregunta al usuario su nombre
const nombre = prompt('Cual es tu nombre');

//Coge el nombre y lo muestra en pantalla
document.querySelector('.contenido').innerHTML = `${nombre} esta aprendiendo JavaScript Moderno`;

