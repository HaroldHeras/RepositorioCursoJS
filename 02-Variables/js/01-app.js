/** Antes se creaba una variable ocn la palabra reservada "var". Se sigue pudiendo usar */


//Incializar una variable con un valor
var producto = "Monitor de 24 pulgadas";
console.log(producto);

//Las variables se pueden reasignar
producto = 'Monitor de 9 pulgadas';
console.log(producto);

/*JavaScript no es un lenguaje tipado. No hace falta poner el tipo de dato al principio y se puede poner 
 o cambiar al tipo de dato que queramos*/
producto = 20;
console.log(producto);

//Se pueden inicializar sin valor y darselo despues
var disponible;

disponible = true;

disponible = false


//Inicializar multiples variables

var categoria = "Computadoras",
    marca = "Marca Famosa",
    calificacion = 5;

//No se puede asignar un nombre de variable que empiece con un n√∫mero.

var 01nombre; //Esto no se puede

var _01nombre; //Esto si que se puede

//No puede haber espacios en los nombres de variables

var nombre producto; //esto no se puede
var nombreProducto; //Esto si se puede


