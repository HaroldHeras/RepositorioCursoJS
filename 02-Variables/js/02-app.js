/*Ahora se crean las variables con las palabras reservadas "let" y "const"*/

let producto = "tablet";

producto = "Monitor";

producto = 20;

producto = true;

producto = null;

console.log(producto);

let precio;

console.log(precio);