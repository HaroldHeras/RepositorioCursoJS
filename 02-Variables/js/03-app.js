/*Las varaibels creadas con la palalbra reservada "const" son constantes y no se les puede reasignar 
el valor. Además, siempre hay que inicializarlas con un valor. No se pueden inicializar y darles un valor después*/


const productoUno; //esto no se puede

const productoDos = "Monitor"; //esto se puede
