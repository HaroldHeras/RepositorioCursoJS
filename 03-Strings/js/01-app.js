/**Vemos las tres formas de crear un String */


const producto = "Monitor de 20 pulgadas";          //Puede ser con comillas dobles

const producto2 = String('Monitor de 24 ');         //Con comillas simples      

const producto3 = new String("Monitor de 27\" ");         //O con dobles, pero si queremos meter otras dobles en el mensaje, metemos unas barras antes de las comillas

console.log(producto);
console.log(producto2);
console.log(producto3);