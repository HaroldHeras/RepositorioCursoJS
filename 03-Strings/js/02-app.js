

const producto = 'Monitor de 20 pulgadas';

console.log(producto);              //Imprimimos en consola el valor de la variable producto

console.log(producto.length);       //Imprimimos en consola el número de letars del valor de la variable producto


console.log(producto.indexOf("pulgadas"));          //Nos sale en que posicion de la cadena de texto aparece la palabra que escribimos entre paréntesis.

console.log(producto.includes("Tablet"));  //Busca una palabra en esa variable y nos dice si está o no. 