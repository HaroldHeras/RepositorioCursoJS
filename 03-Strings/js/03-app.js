/*Concatenación de Strings*/


const producto = "Monitor de 20 pulgadas";              

console.log(producto);                                  

const precio = " 30 usd";

console.log(producto.concat(precio));                   //Podemos concatenar variables con el método .concat y añadir otra variable.

console.log(producto.concat(" en descuento"));          //Con el método .concat y escribir directamente el mensaje.

console.log(producto + " solo hoy.");                   //Sin el método .concat, usando el simbolo + o una coma.

console.log(producto, " muy barato.");



//Importante para esta forma: El mensaje tiene que ir entre comillas inversas `Mensaje`. (En el teclado, al lado de la letra P)
console.log(`El producto ${producto} tiene un precio de ${precio}`);        //Escribiendo el mensaje directamente y añadiendo las variables en el mensaje.