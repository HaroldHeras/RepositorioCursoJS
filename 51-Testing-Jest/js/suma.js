
//Funci√≥n que vamos a exportar para poder ser usada en el archivo de tests



export function suma(a,b){

    return a+b;

}