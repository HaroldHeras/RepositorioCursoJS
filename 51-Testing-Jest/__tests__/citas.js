
//Probamos el tetsing con una clase y sus métodos, importando esa clase.
//Usamos los Snapshots para ver lo que hay en el interior de la clase despues de la ejecución del test. Para ello usamos el comando "npx jest -u" para que lo sobreescriba en cada ejecución, sino dará fallo


import {Citas} from "../js/classes/Citas.js";


describe("Probar la clase de Citas", ()=>{

    const citas = new Citas();                  //Creamos un objeto Citas desde fuera de los tests, para poder usarlo en todos

    const id = Date.now();                      //Cramos el id con la fecha en milisegundos desde fuera, para que todos tengan el mismo id


    test("Agregar una nueva cita", ()=>{        //Evaluamos que se cree una cita nueva

        const citaObj = {                       //Creamos un objeto cita
            id,
            mascota: 'Hook',
            propietario: 'Harold',
            telefono: '123456789',
            fecha: '10-12-2020',
            hora:'10:30',
            sintomas: 'Solo duerme'
        }


        citas.agregarCita(citaObj);             //Usamos el método para agregar la cita al objeto Citas


        expect(citas).toMatchSnapshot();        //la primera vez que se ejecuta, valdrá con hacer "npm t". Crea el Snapshot si se ha ejecutado bien el método. Ahí podremos ver si guarda bien la cita

    })



    test("Actualizar Cita", ()=>{

        const citaActualizada = {                   //Creamos una cita diferente, con el mismo id, para actualizar la que ya hay.
            id,
            mascota: 'Otto',
            propietario: 'Harold',
            telefono: '123456789',
            fecha: '10-12-2020',
            hora:'10:40',
            sintomas: 'Solo duerme'
        }


        citas.editarCita(citaActualizada);          //Usamos el método para editar la cita

        expect(citas).toMatchSnapshot();            //Tendremos que usar el comando "npx jest -u" para que sobreescriba el snapshot y ver lo que hay ahora dentro del objeto Citas

    })



    test("Eliminar Cita", ()=>{

        citas.eliminarCita(id);                     //Usamos el método para eliminar la cita que tenga ese id


        expect(citas).toMatchSnapshot();            //Tendremos que usar el comando "npx jest -u" para que sobreescriba el snapshot y ver lo que hay ahora dentro del objeto Citas


    })


})