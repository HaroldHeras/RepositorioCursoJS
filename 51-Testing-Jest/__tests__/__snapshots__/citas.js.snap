// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Probar la clase de Citas Actualizar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:40",
      "id": 1750165032931,
      "mascota": "Otto",
      "propietario": "Harold",
      "sintomas": "Solo duerme",
      "telefono": "123456789",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1750165032931,
      "mascota": "Hook",
      "propietario": "Harold",
      "sintomas": "Solo duerme",
      "telefono": "123456789",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": [],
}
`;
